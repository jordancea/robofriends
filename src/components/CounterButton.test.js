import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CounterButton from "./CounterButton";

it("should render the CounterButton component", () => {
  const mockColor = "red";
  expect(
    toJson(shallow(<CounterButton color={mockColor} />))
  ).toMatchSnapshot();
});

it("should correctly increment the CounterButton component", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.find('[id="counter"]').simulate("click"); //simulate click event on the button
  expect(wrapper.state()).toEqual({ count: 2 });
  expect(wrapper.props().color).toEqual("red");
});
