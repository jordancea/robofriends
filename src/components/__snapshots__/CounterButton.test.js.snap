// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the CounterButton component 1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  1
</button>
`;
